File: src/server/src/main/java/com/project/infrastructure/data/EquipmentRepository.java
package com.project.infrastructure.data;

import com.project.domain.workout.Equipment;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.CrudRepository;

import java.util.List;
import java.util.UUID;

@Repository
public interface EquipmentRepository extends CrudRepository<Equipment, UUID> {
    List<Equipment> findByIdIn(List<UUID> ids);
}

File: src/server/src/main/kotlin/com/project/infrastructure/data/EquipmentRepository.kt
package com.project.infrastructure.data

import com.project.domain.workout.Equipment
import io.micronaut.data.annotation.Repository
import io.micronaut.data.repository.CrudRepository
import java.util.UUID

@Repository
interface EquipmentRepository : CrudRepository<Equipment, UUID> {
    fun findByIdIn(ids: List<UUID?>): List<Equipment>
}
